### PR TITLE
Update the Xcode version to support Xcode 13 beta 4

### PR DESCRIPTION
**Pull request by shahmishal**
(cherry picked from commit 5363a80aebcddbaf0c101ef1ce948d69fd78b13e)
cherry-picked test fixes for COW checks from main branch (https://github.com/apple/swift/pull/37945)

### CODE REVIEW

**Comment by shahmishal** [comment]
@swift-ci test new Xcode

**Comment by shahmishal** [comment]
https://github.com/apple/swift-driver/pull/787
@swift-ci test new Xcode

**Comment by shahmishal** [comment]
@swift-ci test new xcode

**Comment by shahmishal** [comment]
apple/swift-driver#787
@swift-ci test new Xcode

**Comment by shahmishal** [comment]
apple/swift-driver#787
@swift-ci test new Xcode

**Comment by shahmishal** [comment]
apple/swift-driver#787
@swift-ci test new Xcode

**Comment by shahmishal** [comment]
apple/swift-driver#787
@swift-ci test new Xcode

**Comment by shahmishal** [comment]
@swift-ci test

File: test/lit.cfg [hunk 1]
```diff
@@ -2128,6 +2128,20 @@ config.substitutions.append(('%FileCheck', run_filecheck))
...
 config.substitutions.append(('%import-libdispatch', getattr(config, 'import_libdispatch', '')))
 
+# Disabe COW sanity checks in the swift runtime by default.
+# (But it's required to set this environment variable to something)
+config.environment['SWIFT_DEBUG_ENABLE_COW_CHECKS'] = 'false'
+
+# Add this to the command which runs an executable to enable COW checks in the swift runtime.
+config.substitutions.append(('%enable-cow-checking', TARGET_ENV_PREFIX + 'SWIFT_DEBUG_ENABLE_COW_CHECKS=true;'))
+
+# Disabe COW sanity checks in the swift runtime by default.
```
Comment (tbkka) [inline test/lit.cfg:2138]: This is duplicated?

File: test/lit.cfg [hunk 2]
```diff
@@ -2128,6 +2128,20 @@ config.substitutions.append(('%FileCheck', run_filecheck))
 config.substitutions.append(('%raw-FileCheck', shell_quote(config.filecheck)))
 config.substitutions.append(('%import-libdispatch', getattr(config, 'import_libdispatch', '')))
 
+# Disabe COW sanity checks in the swift runtime by default.
+# (But it's required to set this environment variable to something)
+config.environment['SWIFT_DEBUG_ENABLE_COW_CHECKS'] = 'false'
+
+# Add this to the command which runs an executable to enable COW checks in the swift runtime.
+config.substitutions.append(('%enable-cow-checking', TARGET_ENV_PREFIX + 'SWIFT_DEBUG_ENABLE_COW_CHECKS=true;'))
+
```
Comment (shahmishal) [inline test/lit.cfg:2137]:
```suggestion
```

**Comment by shahmishal** [comment]
apple/swift-driver#787
@swift-ci test new Xcode

**Comment by shahmishal** [comment]
@swift-ci test

**Comment by shahmishal** [comment]
apple/swift-driver#787
@swift-ci build toolchain new Xcode

**Comment by shahmishal** [comment]
apple/swift-driver#787
@swift-ci test new Xcode

**Comment by shahmishal** [comment]
apple/swift-driver#787
@swift-ci test new Xcode

**Comment by shahmishal** [comment]
apple/swift-driver#787
@swift-ci build toolchain new Xcode

**Comment by shahmishal** [comment]
apple/swift-driver#787
@swift-ci test new Xcode

**Comment by shahmishal** [comment]
apple/swift-driver#787
@swift-ci test

**Comment by shahmishal** [comment]
apple/swift-driver#787
@swift-ci test

**Comment by shahmishal** [comment]
apple/swift-driver#787
@swift-ci test new Xcode

**Comment by shahmishal** [comment]
apple/swift-driver#787
@swift-ci build toolchain new Xcode

**Comment by shahmishal** [comment]
apple/swift-driver#787
@swift-ci build toolchain new Xcode

**Comment by shahmishal** [comment]
apple/swift-driver#787
@swift-ci test new Xcode

**Comment by shahmishal** [comment]
apple/swift-driver#787
@swift-ci test

**Comment by shahmishal** [comment]
apple/swift-driver#787
@swift-ci test new Xcode

**Comment by shahmishal** [comment]
@swift-ci test windows

**Review by eeckstein — APPROVED** [review]
LGTM

**Comment by shahmishal** [comment]
**Information related to the build failure is unidentified.**
Please check the respective logs for more information
Build Log: [Test](https://ci.swift.org/job/swift-PR-macOS-new-xcode/121/)

**Comment by shahmishal** [comment]
**Information related to the build failure is unidentified.**
Please check the respective logs for more information
Build Log: [Test](https://ci.swift.org/view/Pull%20Request/job/swift-PR-macOS-new-xcode/126/consoleText)

**Comment by shahmishal** [comment]
**Errors Detected:** Regression test failed
<details>
<summary>Failed Command</summary>
/usr/local/bin/cmake --build /Users/buildnode/jenkins/workspace/swift-PR-macOS-new-xcode/branch-main/buildbot_incremental/swift-macosx-x86_64 -- -j24 -v check-swift-validation-iphonesimulator-x86_64
</details>


```
 Swift-Unit :: SourceKit/SwiftLang/./SourceKitSwiftLangTests/EditTest.DiagsAfterEdit 
```
Build Log: [Test](https://ci.swift.org/view/Pull%20Request/job/swift-PR-macOS-new-xcode/126/consoleText)

**Comment by shahmishal** [comment]
@swift-ci test windows

**Pull request merged**